fl4health/datasets/skin_cancer/preprocess_skin.py

```diff
@@ -0,0 +1,281 @@
+    with open(path, "w", encoding="utf-8") as file:
+        json.dump(data, file, indent="\t")
+
+
+if __name__ == "__main__":
+
+    data_path = "fl4health/datasets/skin_cancer"
+
+    # Load GT file of ISIC-19
+    ISIC_2019_path = f"{data_path}/ISIC_2019"
+    ISIC_csv_path = os.path.join(ISIC_2019_path, "ISIC_2019_Training_GroundTruth.csv")
+    ISIC_df = pd.read_csv(ISIC_csv_path)
+
+    # Load metadata file of HAM10000
+    HAM_10000_path = f"{data_path}/HAM10000"
+    HAM_csv_path = os.path.join(HAM_10000_path, "HAM10000_metadata.csv")
+    HAM_df = pd.read_csv(HAM_csv_path)
+
+    # Delete duplicate images in ISIC-19
+    ISIC_meta = pd.read_csv(os.path.join(ISIC_2019_path, "ISIC_2019_Training_Metadata.csv"))
+    barcelona_list = [i for i in ISIC_meta["lesion_id"].dropna() if "BCN" in i]
+    barcelona_core = ISIC_meta[ISIC_meta["lesion_id"].isin(barcelona_list)]
+
+    core_2019 = ISIC_df[ISIC_df["image"].isin(barcelona_core["image"])]
+    core_2019.to_csv(os.path.join(ISIC_2019_path, "ISIC_2019_core.csv"), mode="w")
+
+    # Split rosendahl and vienna in HAM10000
+    rosendahl_data = HAM_df[HAM_df["dataset"] == "rosendahl"]
+    rosendahl_data.to_csv(os.path.join(HAM_10000_path, "HAM_rosendahl.csv"), mode="w")
+    vienna_data = HAM_df[HAM_df["dataset"] != "rosendahl"]
+    vienna_data.to_csv(os.path.join(HAM_10000_path, "HAM_vienna.csv"), mode="w")
+
+    # Load PAD-UFES-20
+    PAD_UFES_20_path = f"{data_path}/PAD-UFES-20"
+    PAD_UFES_20_csv_path = os.path.join(PAD_UFES_20_path, "metadata.csv")
+    PAD_UFES_20 = pd.read_csv(PAD_UFES_20_csv_path)
+
+    # Load Derm7pt
+    Derm7pt_path = f"{data_path}/Derm7pt"
+    Derm7pt_csv_path = os.path.join(Derm7pt_path, "meta/meta.csv")
+    Derm7pt = pd.read_csv(Derm7pt_csv_path)
+
+    derm7pt_labelmap = {
+        "lentigo": "MISC",
+        "melanosis": "MISC",
+        "miscellaneous": "MISC",
+    }
+
+    MISC_list = ["lentigo", "melanosis", "miscellaneous"]
+    Derm7pt_core = Derm7pt[~Derm7pt["diagnosis"].isin(MISC_list)]
+    Derm7pt_core.to_csv(os.path.join(Derm7pt_path, "meta/meta_core.csv"))
+
+    ###########################################################################
+    # Unify the naming of labels in the below section
+
+    official_columns = ["MEL", "NV", "BCC", "AK", "BKL", "DF", "VASC", "SCC"]
+
+    # Client: Barcelona
+    ###########################################################################
+    ISIC_2019_data_path = f"{data_path}/ISIC_2019/ISIC_2019_Training_Input"
+    ISIC_csv_path = os.path.join(ISIC_2019_path, "ISIC_2019_core.csv")
+    Barcelona_df = pd.read_csv(ISIC_csv_path)
+    Barcelona_new = Barcelona_df[["image", "MEL", "NV", "BCC", "AK", "BKL", "DF", "VASC", "SCC", "UNK"]]
+    preprocessed_data: Dict[str, Any] = {
+        "columns": official_columns,
+        "original_columns": official_columns,
+        "data": [],
+    }
+
+    for i in range(len(Barcelona_new)):
+        temp = list(Barcelona_new.loc[i].values[:-1])
```

Comment:
A comment on why we're leaving off the last piece of data would be useful? That is, why `[:-1]`

Reply:
added a comment, essentially there was an additional column we dont use, so we are ignoring those